juju/provisioner.py

```diff
@@ -12,12 +12,12 @@
 
 
 arches = [
-    [re.compile("amd64|x86_64"), "amd64"],
-    [re.compile("i?[3-9]86"), "i386"],
-    [re.compile("(arm$)|(armv.*)"), "armhf"],
-    [re.compile("aarch64"), "arm64"],
-    [re.compile("ppc64|ppc64el|ppc64le"), "ppc64el"],
-    [re.compile("ppc64|ppc64el|ppc64le"), "s390x"],
+    [re.compile(r"amd64|x86_64"), "amd64"],
+    [re.compile(r"i?[3-9]86"), "i386"],
+    [re.compile(r"(arm$)|(armv.*)"), "armhf"],
+    [re.compile(r"aarch64"), "arm64"],
+    [re.compile(r"ppc64|ppc64el|ppc64le"), "ppc64el"],
+    [re.compile(r"ppc64|ppc64el|ppc64le"), "s390x"],
```

Comment:
I know this was in the code previously (sans 'r'), but why s390 == ppc?  Surely the regex should have been `'s390|s390x'`.

Reply:
That is a good question.  I have no idea.  I'm not sure what the correct values should be.